dissect/util/feature.py

```diff
@@ -0,0 +1,54 @@
+import functools
+import os
+from enum import Enum
+from typing import Callable, Optional
+
+
+# Register feature flags in a central place to avoid chaos
+class Feature(Enum):
+    ADVANCED = "advanced"
+    LATEST = "latest"
+    BETA = "beta"
+
+
+# Defines the default flags (as strings)
+DISSECT_FEATURES_DEFAULT = "latest"
+
+# Defines the environment variable to read the flags from
+DISSECT_FEATURES_ENV = "DISSECT_FEATURES"
+
+
+class FeatureException(RuntimeError):
+    pass
+
+
+@functools.cache
+def feature_flags() -> list[Feature]:
+    return [Feature(name) for name in os.getenv(DISSECT_FEATURES_ENV, DISSECT_FEATURES_DEFAULT).split("/")]
+
+
+@functools.cache
+def feature_enabled(feature: Feature) -> bool:
+    return feature in feature_flags()
+
+
+def feature_disabled_stub() -> None:
+    raise FeatureException("This feature has been disabled.")
+
+
+def feature(flag: Feature, alternative: Optional[Callable] = feature_disabled_stub) -> Callable:
+    """
```

Comment:
```suggestion
    """Feature flag decorator allowing you to guard a function behind a feature flag.

```
Or something along those lines. This sentence is pretty bad, but there should be a first-line docstring that explains the purpose.